src/pybamm/models/full_battery_models/base_battery_model.py

```diff
@@ -513,59 +546,71 @@ def __init__(self, extra_options):
                 )
 
         # Options not yet compatible with contact resistance
-        if options["contact resistance"] == "true":
-            if options["operating mode"] == "explicit power":
+        _ensure_tuple(options["contact resistance"])
+        if "true" in options["contact resistance"]:
+            _ensure_tuple(options["operating mode"])
+            if "explicit power" in options["operating mode"]:
                 raise NotImplementedError(
                     "Contact resistance not yet supported for explicit power."
                 )
-            if options["operating mode"] == "explicit resistance":
+            if "explicit resistance" in options["operating mode"]:
                 raise NotImplementedError(
                     "Contact resistance not yet supported for explicit resistance."
                 )
 
         # Options not yet compatible with particle-size distributions
-        if options["particle size"] == "distribution":
+        _ensure_tuple(options["particle size"])
+        if "distribution" in options["particle size"]:
+            _ensure_tuple(options["lithium plating porosity change"])
             if options["lithium plating porosity change"] != "false":
                 raise NotImplementedError(
-                    "Lithium plating porosity change not yet supported for particle-size"
-                    " distributions."
+                    "Lithium plating porosity change not yet supported for particle-size distributions."
                 )
+
+            _ensure_tuple(options["heat of mixing"])
```

Comment:
Reading them as they come, I feel it would be nicer for readability if all the `_ensure_tuple()` calls would come before conditional statements.